docs/SPEC.md

```diff
@@ -1169,10 +1144,12 @@ All such values must have a corresponding builtin statement to declare the exist
     BuiltinStatement = "builtin" identifier ":" TypeExpression .
     TypeExpression   = MonoType ["where" Constraints] .
 
-    MonoType     = Tvar | BasicType | ArrayType | RecordType | FunctionType .
+    MonoType     = Tvar | BasicType | ArrayType | StreamType | VectorType | RecordType | FunctionType .
     Tvar         = "A" … "Z" .
     BasicType    = "int" | "uint" | "float" | "string" | "bool" | "time" | "duration" | "bytes" | "regexp" .
     ArrayType    = "[" MonoType "]" .
+    StreamType    = "stream" "[" MonoType "]" .
+    VectorType    = "vector" "[" MonoType "]" .
```

Comment:
What about:
```suggestion
    StreamType    = "stream" ArrayType .
    VectorType    = "vector" ArrayType .
```

?

Reply:
I think that conflates the concepts of Stream and vector types. StreamType is not a stream of arrays rather it is its own thing. Same for vector types. So while its an equivalent grammar the semantics are a bit confusing using ArrayType.

Reply:
We split Array and Stream purposefully since Stream represents a collection without a known size. Ref: https://github.com/influxdata/flux/issues/4344